serialize.go

```diff
@@ -138,13 +138,23 @@ func DeserializeArrayNonEmptyDomain(a *Array, buffer *Buffer, serializationType
 
 	if isEmpty == 1 {
 		return nil, true, nil
-	} else {
-		nonEmptyDomains, err := makeNonEmptyDomain(domain, tmpDomain)
+	}
+
+	nonEmptyDomains := make([]NonEmptyDomain, 0)
+	for i := uint(0); i < ndims; i++ {
+		dimension, err := domain.DimensionFromIndex(i)
+		if err != nil {
+			return nil, false, err
+		}
+
+		nonEmptyDomain, err := getNonEmptyDomainForDim(dimension, tmpDomain)
```

Comment:
Same as above you will need to iterate through `tmpDomain` here, else you'll end up with the same values.